disk/disk_windows.go

```
@@ -92,109 +92,118 @@ func UsageWithContext(_ context.Context, path string) (*UsageStat, error) {
+	partitionStats = processVolumesMountedAsFolders(ctx, partitionStats, processedPaths, &warnings)
+	return partitionStats, warnings.Reference()
+}
 
-			for _, mount := range mounts {
-				if _, ok := processedPaths[mount]; ok {
-					continue
-				}
-				if partitionStat, warning := buildPartitionStat(mount); warning == nil {
-					select {
-					case retChan <- partitionStat:
-					case <-quitChan:
-						return
-					}
-				} else {
-					warnings.Add(warning)
-				}
+func processVolumesMountedAsFolders(ctx context.Context, partitionStats []PartitionStat, processedPaths map[string]struct{}, warnings *Warnings) []PartitionStat {
+	// Get volumes without drive letters (ex: mounted folders with no drive letter)
+	volNameBuf := make([]uint16, maxVolumeNameLength)
+	nextVolHandle, _, firstVolumeErr := procFindFirstVolumeW.Call(
+		uintptr(unsafe.Pointer(&volNameBuf[0])),
+		uintptr(maxVolumeNameLength)) // The max length of this should be 50 (it's GUID based string), but I set to the max allowed volume name length
+	if windows.Handle(nextVolHandle) == windows.InvalidHandle {
+		warnings.Add(fmt.Errorf("failed to get first-volume: %w", firstVolumeErr))
+		return partitionStats
+	}
+	defer procFindVolumeClose.Call(nextVolHandle)
+	contextCancelMessage := "context done while processing volumes mounted as folders: %w"
+	for {
+		select {
+		case <-ctx.Done():
+			warnings.Add(fmt.Errorf(contextCancelMessage, ctx.Err()))
+			return partitionStats
+		default:
+			partitionStats = getPartStatFromVolumeName(ctx, volNameBuf, warnings, processedPaths, partitionStats)
+
+			if ctx.Err() != nil {
+				warnings.Add(fmt.Errorf(contextCancelMessage, ctx.Err()))
+				return partitionStats
 			}
 
-			volNameBuf = make([]uint16, volumeNameBufferLength)
-			if volRet, _, err := procFindNextVolumeW.Call(
+			volNameBuf = make([]uint16, maxVolumeNameLength)
+			volRet, _, nextVolumeErr := procFindNextVolumeW.Call(
```

It's paranoid but I don't see a limit in the documentation, I would put a hard limit on the number of iteration we do here, we can return an error if we find more than let's say 200 mount points (random number we can choose what we think is best)

---

the context is already taking care of the stopping condition so it's ok